manifests/init.pp

```diff
@@ -201,10 +208,17 @@
 
   ensure_packages('katello-debug')
 
-  if ($pulp_master or $pulp) and $facts['os']['release']['major'] != '7' {
+  if $enable_pulp2 and $facts['os']['release']['major'] != '7' {
     fail('Pulp 2 is only supported on CentOS 7')
   }
 
+  class { 'foreman_proxy::plugin::pulp':
+    enabled          => $enable_pulp2,
+    pulpnode_enabled => $enable_pulp2_mirror,
```

Comment:
This logic is incorrect. In Pulp 2 there was one plugin for pulp and one for pulpnode. Enabling both at the same time is illogical and probably confuses Katello.
```suggestion
    enabled          => $enable_pulp2 and !$enable_pulp2_mirror,
    pulpnode_enabled => $enable_pulp2 and $enable_pulp2_mirror,
```